Source/Diagnostics/ReducedDiags/RawEFieldReduction.H

```diff
@@ -0,0 +1,275 @@
+                [=] AMREX_GPU_DEVICE (int i, int j, int k) ->ReduceTuple
+                {
+                    // Shift x, y, z position based on index type 
+                    amrex::Real fac_x = (1._rt - Ez_nodalType[0]) * dx[0] * 0.5_rt;
+                    amrex::Real x = i * dx[0] + real_box.lo(0) + fac_x;
+#if (AMREX_SPACEDIM==2)
+                    amrex::Real y = 0._rt;
+                    amrex::Real fac_z = (1._rt - Ez_nodalType[1]) * dx[1] * 0.5_rt;
+                    amrex::Real z = j * dx[1] + real_box.lo(1) + fac_z;
+#else
+                    amrex::Real fac_y = (1._rt - Ez_nodalType[1]) * dx[1] * 0.5_rt;
+                    amrex::Real y = j * dx[1] + real_box.lo(1) + fac_y;
+                    amrex::Real fac_z = (1._rt - Ez_nodalType[2]) * dx[2] * 0.5_rt;
+                    amrex::Real z = k * dx[2] + real_box.lo(2) + fac_z;
+#endif
+                    return reduction_function_parser(x,y,z)*Ez_arr(i,j,k);
+                });
+            }
+
+            amrex::Real reducedEx_value = amrex::get<0>(reduceEx_data.value());
+            amrex::Real reducedEy_value = amrex::get<0>(reduceEy_data.value());
+            amrex::Real reducedEz_value = amrex::get<0>(reduceEz_data.value());
+
+            // MPI Reduce
+            // MPI reduce
+            if (std::is_same<ReduceOp, amrex::ReduceOpMax>::value)
+            {
+                amrex::ParallelDescriptor::ReduceRealMax(reducedEx_value);
+                amrex::ParallelDescriptor::ReduceRealMax(reducedEy_value);
+                amrex::ParallelDescriptor::ReduceRealMax(reducedEz_value);
+            }
+            if (std::is_same<ReduceOp, amrex::ReduceOpMin>::value)
+            {
+                amrex::ParallelDescriptor::ReduceRealMin(reducedEx_value);
+                amrex::ParallelDescriptor::ReduceRealMin(reducedEy_value);
+                amrex::ParallelDescriptor::ReduceRealMin(reducedEz_value);
+            }
+            if (std::is_same<ReduceOp, amrex::ReduceOpSum>::value)
+            {
+                amrex::ParallelDescriptor::ReduceRealSum(reducedEx_value);
+                amrex::ParallelDescriptor::ReduceRealSum(reducedEy_value);
+                amrex::ParallelDescriptor::ReduceRealSum(reducedEz_value);
+            // If reduction operation is a sum, multiply the value by the cell volume so that the
+            // result is the integral of the function over the simulation domain.
+                if (integral_type == 0) {
+
+#if (AMREX_SPACEDIM==2)
+                reducedEx_value *= dx[0]*dx[1];
+                reducedEy_value *= dx[0]*dx[1];
+                reducedEz_value *= dx[0]*dx[1];
+#else
+                reducedEx_value *= dx[0]*dx[1]*dx[2];
+                reducedEy_value *= dx[0]*dx[1]*dx[2];
+                reducedEz_value *= dx[0]*dx[1]*dx[2];
+#endif
+                } else {
+#if (AMREX_SPACEDIM==2)
+		amrex::Real length = m_surface_normal[0]*dx[1] + m_surface_normal[1]*dx[0];
```

Comment:
can you align the code line here with the ones below

Reply:
I did.